lua/cmake-tools/executors/terminal.lua

```diff
@@ -164,31 +165,49 @@ function terminal.send_data_to_terminal(buffer_idx, cmd, opts)
   elseif opts and opts.win_id >= -1 then
     -- The window is not active, we need to create a new buffer
     vim.cmd(":" .. opts.split_direction .. " " .. opts.split_size .. "sp") -- Split
-    vim.api.nvim_win_set_buf(0, buffer_idx)
+    vim.api.nvim_win_set_buf(0, buffer_idx) -- Set buffer to newly created window
   else
     -- log.error("Invalid window Id!")
     -- do nothing
   end
 
-  if opts and not (opts.focus_on_launch_terminal or opts.focus_on_main_terminal) then
-    vim.cmd("wincmd p") -- Goes back to previous window: Equivalent to [[ CTRL-W w ]]
+  -- Now, the cmake buffer's window is currently in focus
+
+  if opts and (opts.focus_on_launch_terminal or opts.focus_on_main_terminal) then
+    -- We want to focus on the newly set terminal
+    vim.api.nvim_set_current_win(opts.win_id)
   elseif opts and opts.start_insert then
+    -- We want to focus on the newly set terminal and enter start_insert
     vim.api.nvim_set_current_win(opts.win_id)
     vim.cmd("startinsert")
   else
-    vim.api.nvim_set_current_win(opts.win_id)
+    -- We want to focus on our currently focused window and not ther cmake terminal
+    local name = vim.api.nvim_buf_get_name(vim.api.nvim_win_get_buf(0))
+    local basename = vim.fn.fnamemodify(name, ":t")
+    if opts and (basename:sub(1, #opts.prefix) == opts.prefix) then -- If currently focused buffer is cmake buffer then ...
+      -- Now we check again if the buffer needs to be focused as the user might be scrolling
+      -- a cmake buffer and execute a :CMakeCommand, so we do not want to move their
+      -- cursor out of the cmake buffer, as it can be annoying
+      if opts and (opts.focus_on_launch_terminal or opts.focus_on_main_terminal) then
```

Comment:
Do you forger a `not`?
To my understanding, it should be `if opts and not (opts.focus_on_launch_terminal or opts.focus_on_main_terminal)`

Reply:
Yes, this is better. Updated.